src/pages/ReportSettingsPage.js

```diff
@@ -65,6 +66,19 @@ class ReportSettingsPage extends Component {
         ];
     }
 
+    /**
+     * @param {Object} report - the given report we're viewing settings for
+     * @param {Object|null} policy - the workspace the report is on, null if the user isn't a member of the workspace
+     * @returns {Boolean}
+     */
+    shouldDisablePublicRoomRename(report, policy) {
+        if (!policy || !ReportUtils.isPublicRoom(report)) {
+            return true;
+        }
+
+        return !Policy.isPolicyOwner(policy) && policy.role !== CONST.POLICY.ROLE.ADMIN;
```

Comment:
Looking at the API commands and auth commands it doesn't look like we do any of these checks in auth so we need to also add them there because Auth is the source of truth and we should ensure anyone trying to edit the db has the correct access to edit it.

These checks are fine here as well but shouldn't be the ONLY checks like it is right now, not a blocker for this PR but make sure to create a follow up issue so that auth does the proper checks to ensure only an admin or policy owner can change the name.

Reply:
Sounds good, I'll make a follow-up

Reply:
https://github.com/Expensify/Expensify/issues/267679